pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -618,6 +624,10 @@ func (p *provider) Cleanup(machine *v1alpha1.Machine, _ *cloudprovidertypes.Prov
 		if err := config.manager.UnregisterInstance(*instance.instance.PrivateDnsName); err != nil {
 			return false, fmt.Errorf("failed delete machine %s subscription: %v", machine.Name, err)
 		}
+
+		if err := rhsm.RemoveRHELSubscriptionFinalizer(machine, data.Update); err != nil {
```

Comment:
Same here, cannot we move this in a centralised place to avoid repeating for different providers?

Reply:
Please refer to the comment about :) 
https://github.com/kubermatic/machine-controller/pull/722#discussion_r395774608